internal/pkg/skuba/deployments/ssh/kubernetes.go

```diff
@@ -113,9 +113,16 @@ func kubernetesUpgradeStageOne(t *Target, data interface{}) error {
 		// 1.17 is the last version included in CaaSP4. It's the tipping
 		// point where we changed our packaging.
 		// On 1.17 we can't remove kubernetes-1.17-kubeadm, because it doesn't exist.
-		// Removing kubeadm keeps kubelet alive.
-		// The rest needs to be removed on the next stage.
-		pkgs = append(pkgs, "-patterns-caasp-Node-1.17", "-\"kubernetes-kubeadm<1.18\"", "-caasp-config")
+		// We are removing kubeadm while keeping kubelet alive to its version 1.17.
+		// For the initial migration we need to update crio kubeadm
+		// to 1.18 in stage1, due to conflict resolution: the caasp4
+		// cri-o-kubeadm-criconfig requires kubernetes-kubeadm which is
+		// not provided anymore (when we remove kubernetes-kubeadm, and
+		// because we don't want to have the same provides: on the new
+		// package to avoid upgrade during zypper migration).
+		// we need to remove cri-o in stage2 else 1.17 kubelet could
+		// complain about cri-runtime being absent.
+		pkgs = append(pkgs, "-patterns-caasp-Node-1.17", "-\"kubernetes-kubeadm<1.18\"", "-caasp-config", "-cri-o-kubeadm-criconfig")
```

Comment:
oh I didn't know zypper was capable to understand a syntax such as `-kubernetes-kubeadm<1.18` :wink:

Reply:
Magic! https://media.giphy.com/media/NmerZ36iBkmKk/giphy.gif